drivers/include/motor_driver.h

```diff
@@ -0,0 +1,220 @@
+ *  ________________________
+ * | DIR | BRAKE | BEHAVIOR |
+ * |------------------------|
+ * |  0  |   0   |    CW    |
+ * |  0  |   1   |   BRAKE  |
+ * |  1  |   0   |    CCW   |
+ * |  1  |   1   |   BRAKE  |
+ * |_____|_______|__________|
+ *
+ * @endverbatim
+ *
+ * From this truth tables we can extract three direction states :
+ * - CW (ClockWise)
+ * - CCW (Counter ClockWise)
+ * - BRAKE
+ *
+ * BRAKE LOW is functionnaly the same than BRAKE HIGH but some H-bridge only
+ * brake on BRAKE HIGH due to hardware.
+ * In case of single direction GPIO, there is no BRAKE, PWM duty cycle is set
+ * to 0.
+
+ * @{
+ * @file
+ * @brief       High-level driver for DC motors
+ *
+ * @author      Gilles DOFFE <g.doffe@gmail.com>
+ */
+
+#ifndef MOTOR_DRIVER_H
+#define MOTOR_DRIVER_H
+
+#include "periph/pwm.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Maximum number of motors by motor driver
+ */
+#ifndef MOTOR_DRIVER_MAX
+#define MOTOR_DRIVER_MAX    (2)
+#endif /* MOTOR_DRIVER_MAX */
+
+#define MOTOR_DRIVER_DEV(x) (x)
+
+/**
+ * @brief Describe DC motor driver modes
+ */
+typedef enum {
+    MOTOR_DRIVER_2_DIRS         = 0,          /**< 2 GPIOS for direction, \
+                                                    handling BRAKE */
+    MOTOR_DRIVER_1_DIR          = 1,          /**< Single GPIO for direction, \
+                                                    no BRAKE */
+    MOTOR_DRIVER_1_DIR_BRAKE    = 2           /**< Single GPIO for direction, \
+                                                    Single GPIO for BRAKE */
+} motor_driver_mode_t;
+
+/**
+ * @brief Describe DC motor driver brake modes
+ */
+typedef enum {
+    MOTOR_BRAKE_LOW     = 0,        /**< Low stage brake */
+    MOTOR_BRAKE_HIGH    = 1,        /**< High stage brake */
+} motor_driver_mode_brake_t;
+
+/**
+ * @brief Describe DC motor direction states
+ */
+typedef enum {
+    MOTOR_CW      = 0,          /**< clockwise */
+    MOTOR_CCW     = 1,          /**< counter clockwise */
+    MOTOR_BRAKE   = 2           /**< brake */
+} motor_direction_t;
+
+/**
+ * @brief Describe DC motor with PWM channel and GPIOs
+ */
+typedef struct {
+    int pwm_channel;            /**< PWM channel the motor is connected to */
+    gpio_t gpio_enable;         /**< GPIO to enable/disable motor */
+    gpio_t gpio_dir0;           /**< GPIO to control rotation direction */
+    gpio_t gpio_dir1_or_brake;  /**< GPIO to control rotation direction */
+    uint8_t gpio_dir_reverse;   /**< flag to reverse direction */
+    uint8_t gpio_enable_invert; /**< flag to set enable GPIO inverted mode */
+    uint8_t gpio_brake_invert;  /**< flag to make brake active low */
+} motor_t;
+
+/**
+ * @brief   Default motor driver type definition
+ */
+typedef unsigned int motor_driver_t;
+
+/**
+ * @brief   Motor callback. It is called at end of motor_set()
+ */
+typedef void (*motor_driver_cb_t)(const motor_driver_t motor_driver,
```

Comment:
This callback is here to do specific board stuff. For example blinking a led when a motor is set, compute or display informations, etc.
In case of native board, used mainly for simulation, it is used to simulate the physical relation between QDEC driver and motor driver.
On native board, actuators driver must create simulation datas according to board sensors related to that actuator.
In a robot for example it could be positionning sensors (QDEC, IMU, triangulation, etc.).
If you use a function called each time after `motor_set` the user code is no more generic, it becomes heavier and is board dependant.
An other solution could be to use a weak function (like `panic_arch()`) but it would be the same function for each motor driver which is less flexible.